sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -0,0 +1,361 @@
+  //     "Failed to get address of clGetDeviceFunctionPointerINTEL function");
+
+  if (!func_ptr) {
+    if (function_pointer_ret)
+      *function_pointer_ret = 0;
+    return PI_INVALID_DEVICE;
+  }
+
+  return PI::cast<pi_result>(func_ptr(PI::cast<cl_device_id>(device),
+                                      PI::cast<cl_program>(program), func_name,
+                                      function_pointer_ret));
+}
+
+// TODO: Remove the 'OclPtr' extension used with the PI_APIs.
+// Forward calls to OpenCL RT.
+#define _PI_CL(pi_api, ocl_api)                                                \
+  decltype(::pi_api) *pi_api##OclPtr = PI::cast<decltype(&::pi_api)>(&ocl_api);
+
+// Platform
+_PI_CL(piPlatformsGet, OCL(piPlatformsGet))
+_PI_CL(piPlatformGetInfo, clGetPlatformInfo)
+// Device
+_PI_CL(piDevicesGet, OCL(piDevicesGet))
+_PI_CL(piDeviceGetInfo, clGetDeviceInfo)
+_PI_CL(piDevicePartition, clCreateSubDevices)
+_PI_CL(piDeviceRetain, clRetainDevice)
+_PI_CL(piDeviceRelease, clReleaseDevice)
+_PI_CL(piextDeviceSelectBinary, OCL(piextDeviceSelectBinary))
+_PI_CL(piextGetDeviceFunctionPointer, OCL(piextGetDeviceFunctionPointer))
+// Context
+_PI_CL(piContextCreate, clCreateContext)
+_PI_CL(piContextGetInfo, clGetContextInfo)
+_PI_CL(piContextRetain, clRetainContext)
+_PI_CL(piContextRelease, clReleaseContext)
+// Queue
+_PI_CL(piQueueCreate, OCL(piQueueCreate))
+_PI_CL(piQueueGetInfo, clGetCommandQueueInfo)
+_PI_CL(piQueueFinish, clFinish)
+_PI_CL(piQueueRetain, clRetainCommandQueue)
+_PI_CL(piQueueRelease, clReleaseCommandQueue)
+// Memory
+_PI_CL(piMemBufferCreate, clCreateBuffer)
+_PI_CL(piMemImageCreate, clCreateImage)
+_PI_CL(piMemGetInfo, clGetMemObjectInfo)
+_PI_CL(piMemImageGetInfo, clGetImageInfo)
+_PI_CL(piMemRetain, clRetainMemObject)
+_PI_CL(piMemRelease, clReleaseMemObject)
+_PI_CL(piMemBufferPartition, clCreateSubBuffer)
+// Program
+_PI_CL(piProgramCreate, OCL(piProgramCreate))
+_PI_CL(piclProgramCreateWithSource, clCreateProgramWithSource)
+_PI_CL(piclProgramCreateWithBinary, clCreateProgramWithBinary)
+_PI_CL(piProgramGetInfo, clGetProgramInfo)
+_PI_CL(piProgramCompile, clCompileProgram)
+_PI_CL(piProgramBuild, clBuildProgram)
+_PI_CL(piProgramLink, clLinkProgram)
+_PI_CL(piProgramGetBuildInfo, clGetProgramBuildInfo)
+_PI_CL(piProgramRetain, clRetainProgram)
+_PI_CL(piProgramRelease, clReleaseProgram)
+// Kernel
+_PI_CL(piKernelCreate, clCreateKernel)
+_PI_CL(piKernelSetArg, clSetKernelArg)
+_PI_CL(piKernelGetInfo, clGetKernelInfo)
+_PI_CL(piKernelGetGroupInfo, clGetKernelWorkGroupInfo)
+_PI_CL(piKernelGetSubGroupInfo, clGetKernelSubGroupInfo)
+_PI_CL(piKernelRetain, clRetainKernel)
+_PI_CL(piKernelRelease, clReleaseKernel)
+// Event
+_PI_CL(piEventCreate, clCreateUserEvent)
+_PI_CL(piEventGetInfo, clGetEventInfo)
+_PI_CL(piEventGetProfilingInfo, clGetEventProfilingInfo)
+_PI_CL(piEventsWait, clWaitForEvents)
+_PI_CL(piEventSetCallback, clSetEventCallback)
+_PI_CL(piEventSetStatus, clSetUserEventStatus)
```

Comment:
piEventSetStatus should be an picl extension, other plugins may not support this feature